ddtrace/tracer/tracer.go

```diff
@@ -216,12 +220,16 @@ func SetUser(s Span, id string, opts ...UserMonitoringOption) {
 // payloadQueueSize is the buffer size of the trace channel.
 const payloadQueueSize = 1000
 
-func newUnstartedTracer(opts ...StartOption) *tracer {
-	c := newConfig(opts...)
+func newUnstartedTracer(opts ...StartOption) (*tracer, error) {
+	c, err := newConfig(opts...)
+	if err != nil {
+		return nil, err
+	}
 	sampler := newPrioritySampler()
 	statsd, err := newStatsdClient(c)
 	if err != nil {
-		log.Warn("Runtime and health metrics disabled: %v", err)
+		log.Error("Runtime and health metrics disabled: %v", err)
+		return nil, err
```

Comment:
I looked into the errors that the newStatsDClient returns, and it looks like it could be a bunch of different errors from all kinds of places, so I would suggest wrapping the error with something like `cound't initialise statsd client: <error>`. You can check what other errors the tracer returns, they will have proper formatting and wording